mautic/mautic
# Fix report email failed stats graph performance

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
This PR increase performance of failed graph email stats of reports.

For example this query:

`SELECT e.id, e.subject as title, count(CASE WHEN es.is_failed THEN 1 ELSE null END) as failed FROM email_stats es LEFT JOIN emails e ON e.id = es.email_id LEFT JOIN leads l ON l.id = es.lead_id LEFT JOIN ip_addresses i ON i.id = es.ip_id LEFT JOIN emails vp ON vp.id = e.variant_parent_id LEFT JOIN categories c ON c.id = e.category_id WHERE (es.date_sent IS NULL OR (es.date_sent BETWEEN '2021-01-31 23:00:00' AND '2021-06-20 21:59:59')) GROUP BY e.id, e.subject HAVING count(CASE WHEN es.is_failed THEN 1 ELSE null END) > 0 ORDER BY failed DESC LIMIT 10
`

> Query took 163.4185 seconds

After patch 

`SELECT e.id, e.subject as title, count(CASE WHEN es.is_failed THEN 1 ELSE null END) as failed FROM email_stats es LEFT JOIN emails e ON e.id = es.email_id LEFT JOIN leads l ON l.id = es.lead_id LEFT JOIN ip_addresses i ON i.id = es.ip_id LEFT JOIN emails vp ON vp.id = e.variant_parent_id LEFT JOIN categories c ON c.id = e.category_id WHERE (es.date_sent IS NULL OR (es.date_sent BETWEEN '2021-01-31 23:00:00' AND '2021-06-20 21:59:59')) AND (es.is_failed = 1) GROUP BY e.id, e.subject HAVING count(CASE WHEN es.is_failed THEN 1 ELSE null END) > 0 ORDER BY failed DESC LIMIT 10`

> Query took 4.1649 seconds



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Just check If email stats failed graph works like before. ON bigger instance you feel free run query above and check performance boost
